test/integration/discovery.go

```diff
@@ -157,6 +288,129 @@ var _ = Describe("Discover Cluster", func() {
 	})
 })
 
+func findEntities(vals []*proto.EntityDTO, condition func(v *proto.EntityDTO) bool, howMany int) []*proto.EntityDTO {
```

Comment:
Do we really need these `howMany` stuff?   It seems that we only need to find the first one element that match the condition and if it exists we can just return `true`, otherwise return `false` , I don't see the point to return a slice here.

Reply:
On line 255 I have to find multiple pods, but since most of the time I am just using it as a way to find one element I like being able to break out of the loop, since I no longer need to iterate over the range. It's more efficient really.